packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts

```diff
@@ -189,7 +190,7 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
         flatData = {
           apiVersion: awsService.config.apiVersion, // For test purposes: check if apiVersion was correctly passed.
           region: awsService.config.region, // For test purposes: check if region was correctly passed.
-          ...flatten(response),
+          ...flatten(response, call.encoding ?? 'utf8'),
```

Comment:
The default is utf8 in node

```suggestion
          ...flatten(response, call.encoding),
```